spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulLifecycle.java

```diff
@@ -81,7 +82,9 @@ protected void register() {
 		Assert.notNull(service.getPort(), "service.port has not been set");
 		String appName = getAppName();
 		service.setId(getServiceId());
-		service.setAddress(properties.getHostname());
+		if(properties.getAddressSource() != AddressSource.AGENT) {
```

Comment:
If you'd rather not change the default that is fine, though I would imagine this would become a very common thing people would override (at least in our organization it would be every single project)